File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        numj = 0
        currw = 0
        furtheste = 0
        for i in range(len(nums) - 1):
            if furtheste >= len(nums) - 1:
                numj += 1
                break
            furtheste = max(furtheste , i + nums[i])
            if i == currw:
                numj += 1
                currw = furtheste
            if currw >= len(nums) - 1:
                break
        return numj


        

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        if(n == 1) return 0;
        int i = 0;
        int j = nums[0];
        int jumps = 1;
        while(j < n - 1){
            int k = j;
            while(i <= j){
                k = max(k, i + nums[i]);
                ++i;
            }
            j = k;
            ++jumps;
        }
        return jumps;
    }
};